fplus-http-server/src/router/application.rs

```diff
@@ -23,12 +23,13 @@ pub async fn create(info: web::Json<CreateApplicationInfo>) -> impl Responder {
 pub async fn single(query: web::Query<ApplicationQueryParams>) -> impl Responder {
     let ApplicationQueryParams { id, owner, repo } = query.into_inner();
     match LDNApplication::load_from_db(id, owner, repo).await {
-        Ok(app_file) => match serde_json::to_string_pretty(&app_file) {
-            Ok(response) => HttpResponse::Ok().body(response),
-            Err(_) => {
+        Ok(app_file) => {
+            if let Ok(response) = serde_json::to_string_pretty(&app_file) {
+                HttpResponse::Ok().body(response)
+            } else {
                 HttpResponse::InternalServerError().body("Failed to serialize success message")
             }
-        },
+        }
         Err(e) => HttpResponse::BadRequest().body(e.to_string()),
     }
 }
```

Comment:
Is there a way to refactor it better? `match` -> `if let` doesn't really give us much, it's still nested code.
Ideally, I'd like to see something like this? Not tested though:
```rust
let app_file = LDNApplication::load_from_db(id, owner, repo).await.map_err(ErrorNotFound)?;
let body = serde_json::to_string_pretty(&app_file).map_err(ErrorInternalServerError)?;
HttpResponse::Ok().body(body)
```
https://docs.rs/actix-web/latest/actix_web/error/fn.ErrorNotFound.html
https://docs.rs/actix-web/latest/actix_web/error/fn.ErrorInternalServerError.html

Reply:
Applies to other instances in this PR as well.